sdk/keyvault/keyvault-keys/src/index.ts

```diff
@@ -419,6 +428,32 @@ export class KeyClient {
     });
   }
 
+  /**
+   * Gets a {@link CryptographyClient} for the given key.
+   *
+   * Example usage:
+   * ```ts
+   * let client = new KeyClient(url, credentials);
+   * // get a cryptography client for a given key
+   * let cryptographyClient = client.getCryptographyClient("MyKey");
+   * ```
+   * @param name - The name of the key used to perform cryptographic operations.
+   * @param version - Optional version of the key used to perform cryptographic operations.
+   * @returns - A {@link CryptographyClient} using the same options, credentials, and http client as this {@link KeyClient}
+   */
+  public getCryptographyClient(keyName: string, keyVersion?: string): CryptographyClient {
+    // The goals of this method are discoverability and performance (by sharing a client and pipeline).
+    // The existing cryptography client does not accept a pipeline as an argument, nor does it expose it.
+    // In order to avoid publicly exposing the pipeline we will access the underlying client
+    // via private variables and set the client to shared client.
+    const keyUrl = new URL(["keys", keyName, keyVersion].filter(Boolean).join("/"), this.vaultUrl);
+    const cryptoClient = new CryptographyClient(keyUrl.toString(), this.credential);
+    if (cryptoClient["remoteProvider"]) {
+      cryptoClient["remoteProvider"]["client"] = this.client;
```

Comment:
Does this really help perf that much? We're still manufacturing an entire generated client instance, but we then throw it away and perhaps immediately trigger a GC. 

Yeah, our working set stays a little smaller, but I think if you're going to do this, you need a way to avoid building the generated client a second time. Can't we just have an undocumented property on CryptographyClientOptions that lets us pass the client directly?

Reply:
Yea that's totally fair. I can pass the client in through the constructor instead. Let me rework this and I'll let you know when it's ready for another review

Reply:
Ok, I reworked the implementation so that we don't create a throwaway client unnecessarily - does this look better or do you think we should add `KeyVaultClient` as a property to CryptographyClientOptions and mark it internal? I did not want to add a property for the generated client but am happy to defer to you.

Reply:
I'm good with this approach